odbc/src/api/bind_col_tests.rs

```diff
@@ -466,4 +403,832 @@ mod unit {
         }
     }
+
+    #[test]
+    fn test_binding_arrays_when_rowset_size_evenly_divides_resultset_size() {
+        // Set up MongoHandle
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        unsafe {
+            // Get Statement
+            let s = (*stmt).as_statement().unwrap();
+
+            // set every value in the array to 0, so we know SQLFetch changed the values when we check later.
+            let num_buffer: *mut std::ffi::c_void = Box::into_raw(Box::new([0i32; 4])) as *mut _;
+            let num_indicator: *mut Len = Box::into_raw(Box::new([0isize; 4])) as *mut Len;
+
+            let word_buffer: *mut std::ffi::c_void = Box::into_raw(Box::new([0u32; 20])) as *mut _;
+            let word_indicator: *mut Len = Box::into_raw(Box::new([0isize; 4])) as *mut Len;
+
+            // In this test, we assume that SQLBindCol has already been run and added columns to bind, so
+            // I add column `1` and `2` to bound_cols.
+            *s.bound_cols.write().unwrap() = Some(map! {
+                1 => BoundColInfo {
+                    target_type: CDataType::SQL_C_SLONG as SmallInt,
+                    target_buffer: num_buffer,
+                    buffer_length: 4, // buffer_length is 4 because an i32 is 4 bytes; therefore, each buffer needs to be 4 bytes long.
+                    length_or_indicator: num_indicator,
+                },
+                2 => BoundColInfo {
+                    target_type: CDataType::SQL_C_WCHAR as SmallInt,
+                    target_buffer: word_buffer,
+                    buffer_length: 20, // buffer_length is 20 because each word is 20 bytes long including the null termination character (WideChar * 5).
+                    length_or_indicator: word_indicator,
+                },
+
+            });
+
+            // set all statement attributes to the correct values.
+            s.attributes.write().unwrap().row_bind_offset_ptr = null_mut();
+            // row_array_size is 4 meaning sqlFetch will fetch and handle the column bindings for 4 rows at a time.
+            s.attributes.write().unwrap().row_array_size = 4;
+            s.attributes.write().unwrap().row_bind_type = BindType::SQL_BIND_BY_COLUMN as usize;
+
+            // row_status_ptr needs 4 slots since the rowset size is 4.
+            s.attributes.write().unwrap().row_status_ptr =
+                Box::into_raw(Box::new([0u16; 4])) as *mut USmallInt;
+            s.attributes.write().unwrap().rows_fetched_ptr =
+                Box::into_raw(Box::new(0usize)) as *mut ULen;
+
+            // create a mongo query with data that corresponds to the bound columns.
+            let mock_query = MongoQuery::new(
```

Comment:
Yeah that's a good idea!

Reply:
I condensed all the stuff that's easy to condense. Further condensing would require more work since I would need to make macros and deal with lifetime stuff. I can continue condensing if that's preferable, but since the test file is already much shorter, I figured I would stop here for now and check if we want to do that.